guestbook.module

```diff
@@ -0,0 +1,177 @@
+ *
+ * @param $value
+ *   Implements hook_load
+ *   contains array('id', 'uid', 'sid', 'date', 'message').
+ * @return
+ *   Return form.
+ */
+function guestbook_page_form($form, $form_state, $value= NULL) {
+  global $user;
+  // Text for editing.
+  $text_edit = NULL;
+
+  // Get html list of messages.
+  $text = guestbook_html();
+
+  $form = [];
+
+  if (isset($value['id'])) {
+    // Get tex for editing.
+    $text_edit = $value['message'];
+    $form['infoById'] = array('#type' => 'value', '#value' => $value);
+  }
+
+  $form['#attached']['css'][] = drupal_get_path('module', 'guestbook') . '/css/guestbook.css';
+
+  $form['message_window'] = [
+    // Show messages if it exists.
+    '#markup' => $text,
+  ];
+
+  // Show area for input text if user is logged.
+  if ($user->uid) {
+    $form['message'] = [
+      '#type' => 'textarea',
+      '#title' => t('message'),
+      '#size' => 1000,
+      '#default_value' => $text_edit,
+    ];
+
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#attributes' => array('class' => array('guestbook-button')),
+      '#value' => isset($value['id']) ? t('Edit') : t('Save'),
+    ];
+
+    // Show or hide the cancel button.
+    if (isset($value['id'])) {
+      $form['cancel_button'] = [
+        '#type' => 'link',
+        '#title' => t('Cancel'),
+        '#href' => 'guestbook-page',
+        '#options' => array('query' => drupal_get_query_parameters()),
+        '#attributes' => array('class' => array('guestbook-button')),
+
+      ];
+    }
+  }
+  return $form;
+}
+
+/**
+ * Validate guestbook_page_form.
+*/
+function guestbook_page_form_validate($form, &$form_state) {
+  if (drupal_strlen($form_state['values']['message']) > 1000 || empty($form_state['values']['message'])) {
+    form_set_error('message', t('Field "@fieldname" must be less than @c chars and cant be empty .',
+      array('@fieldname' => t('message'), '@c' => 1000)));
+  }
+}
+
+/**
+ * Submit for guestbook_page_form.
+ */
+function guestbook_page_form_submit($form, &$form_state) {
+  guestbook_save_edit_db($form_state);
+}
+
+/**
+ * Loads a message info.
+ *
+ * @param $mid
+ *   Integer specifying the message id to load.
+ * @return
+ *   A fully-loaded array of message info ('id', 'uid', 'sid', 'date', 'message').
+ *   id - message id; uid - user id; sid - session id; date - message creation date; message - message text.
+ */
+function mid_load($mid) {
+  $query = db_select('guestbook_drupal', 't')
+    ->fields('t', array('id', 'uid', 'sid', 'date', 'message'))
+    ->condition('t.id', $mid)
+    ->execute();
+  $value = $query->fetchAssoc();
+  return $value;
+}
+
+/*
+ * Implements hook_cron().
+ */
+function guestbook_cron() {
+  guestbook_get_xml_weather();
+}
+
+/*
+ * Implements hook_block_info().
+ */
+function guestbook_block_info() {
+  // Build weather block.
+  $blocks['weather'] = array(
+    'info' => t('The Weather'),
+    'cache' => DRUPAL_CACHE_GLOBAL,
+    'region' => 'sidebar_first',
```

Comment:
you can remove this line and give the user ability in which region the block should be placed manually after module installation.